### PR TITLE
NPS Survey: Styling

### PR DESCRIPTION
**Pull request by mattsherman**
This PR styles the NPS Survey dialog.

nps survey notice

Test:
- To see debug messages, in browse console:
    - `localStorage.setItem('debug', 'calypso:nps-survey');`
- from `http://calypso.localhost:3000/devdocs/blocks/nps-survey`
    - Survey should submit successfully the first time you answer it (the API errors if a score has been provided in the past 4 months)
- from global notice (see #12522)
    - Survey should error (silently; it always will thank the user for submitting -- no need to trouble the user should there be a problem submitting), since the survey name is not yet whitelisted in the API endpoint
- Verify that dismissing survey without answering it calls API (look at debug messages and Redux actions/state)

For now, this is not enabled in any environments. To enable:

```
ENABLE_FEATURES=nps-survey/notice make run
```

To re-trigger the NPS survey notice, in the browser console, type:

```
dispatch( showNpsSurveyNotice() );
```

### CODE REVIEW

**Comment by mattsherman** [comment]
@shaunandrews I used your original mock as the basis for the styling. Please review and let me know if there are any tweaks you'd like to see!

File: client/blocks/nps-survey/index.jsx [hunk 1]
```diff
@@ -71,45 +75,51 @@ class NpsSurvey extends Component {
...
-							I'd rather not answer
+							{ translate( 'I\'d rather not answer' ) }
 						</Button>
 					</div>
 				</div>
 				<div className="nps-survey__thank-you-screen">
-					Thanks for providing your feedback!
-					<div>
+					<div className="nps-survey__thank-you">
+						{ translate( 'Thanks for your feedback!' ) }
```
Comment (a8ci18n) [inline client/blocks/nps-survey/index.jsx]: Hi! I've found a possible matching string that has already been translated **27** times:
`translate( 'Thanks for your feedback.', { context: 'calypso stats survey'} )` *ES Score: 14*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/blocks/nps-survey/index.jsx [hunk 2]
```diff
@@ -71,45 +75,51 @@ class NpsSurvey extends Component {
...
 					</div>
-					<div>
+					<div className="nps-survey__buttons">
 						<Button primary
 							className="nps-survey__finish-button"
 							disabled={ shouldDisableControls }
 							onClick={ this.handleFinishClick }
 						>
-							Finish
+							{ translate( 'Finish', { context: 'verb' } ) }
```
Comment (a8ci18n) [inline client/blocks/nps-survey/index.jsx]: :disappointed: 68 existing translations will be lost with this change.
If the string must change, the following option has already been translated **63** times:
`translate( 'Finish', { context: 'spellcheck'} )` *ES Score: 16*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by shaunandrews** [comment]
screen shot 2017-03-30 at 4 20 19 pm

I know others have looked at this copy and made suggestions, but the wording feels very awkward. It doesn't feel natural. I think its the "Would you" part that I'm tripping up on. Maybe something like "Would you mind answering a question..."

--

screen shot 2017-03-30 at 4 11 41 pm

Looking at the "Unlikely" and "Very Likely" labels again, I wonder if they could be confusing — they don't visually line up with a specific radio button. Maybe we can add some visual cue to help associate the labels with 0 and 10:

default state

--

I also notice the empty space where the button will appear:

screen shot 2017-03-30 at 4 13 34 pm

Lets clean this up in the default state; Once they select a radio the button can transition in:

nps survey button transition

--

The success page is a little underwhelming. I wonder if instead, we close the modal immediately and show a success notice:

success notice

**Comment by mattsherman** [comment]
Thanks for the feedback @shaunandrews -- I will work on incorporating those changes.

I'm not sure about the visual cue for the "Unlikely" and "Likely" labels -- I don't think the visual cues are necessary, though I do see where there could be confusion. I'll give it a try though!

I like the success notice idea instead of the success page.

**Comment by mattsherman** [comment]
@shaunandrews The horizontal spacing is already extremely tight on mobile. I don't think it can accommodate the visual cues for the "Unlikely" and "Very Likely" labels, as it pushes those labels out further horizontally.

screencapture at fri mar 31 07 37 34 edt 2017

File: client/blocks/nps-survey/index.jsx [hunk 3]
```diff
@@ -49,49 +51,57 @@ class NpsSurvey extends Component {
...
 					</div>
-					<div>
+					<div className="nps-survey__buttons">
 						<Button primary
 							className="nps-survey__finish-button"
 							disabled={ this.props.hasAnswered }
 							onClick={ this.handleFinishClick }
 						>
-							Finish
+							{ translate( 'Finish', { context: 'verb' } ) }
```
Comment (a8ci18n) [inline client/blocks/nps-survey/index.jsx]: :disappointed: 68 existing translations will be lost with this change.
If the string must change, the following option has already been translated **63** times:
`translate( 'Finish', { context: 'spellcheck'} )` *ES Score: 16*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/nps-survey-styles)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Comment (yoavf) [inline client/blocks/nps-survey/index.jsx]: 👋 
You can use `Finish` with no context. It's mostly not ambiguous, and we've already got that translated.
(The a8ci18n bot comment here is somewhat broken, but you can see the proper suggestion here: https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/nps-survey-styles)

Comment (mattsherman) [inline client/blocks/nps-survey/index.jsx]: Thanks for the clarification @yoavf -- I'll drop the context.

File: client/blocks/nps-survey/index.jsx [hunk 4]
```diff
@@ -38,75 +40,83 @@ class NpsSurvey extends Component {
...
 
 	handleDismissClick = () => {
 		this.props.submitNpsSurveyWithNoScore( this.props.name );
-		// allow for the state to propagate
+		this.onClose( noop );
+	}
+
+	showThanksNotice = () => {
+		this.props.successNotice(
+			this.props.translate( 'Thanks for your feedback!' ),
```
Comment (a8ci18n) [inline client/blocks/nps-survey/index.jsx]: Hi! I've found a possible matching string that has already been translated **27** times:
`translate( 'Thanks for your feedback.', { context: 'calypso stats survey'} )` *ES Score: 14*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by mattsherman** [comment]
@shaunandrews I have changed the following, based on your feedback:

- Notice prompt text
- Switched from thank you screen to success notice
- Finish button transition (I slightly differed in the transition from what you had mocked, because the dialog re-centers when it changes height, and that felt odd. So, I implemented it such that the dialog maintains a constant height.

I did not end up changing the "Unlikely" and "Very Likely" labels, because:

- On mobile there would be horizontal space issues
- With translations, there would be layout issues, as if the translations were much longer, things would look odd

We can iterate further on these labels if desired in a future PR after this one is merged.

/cc @mjuhasz @jonathansadowski @jblz This is is now ready for review.

**Comment by jonathansadowski** [comment]
> I know others have looked at this copy and made suggestions, but the wording feels very awkward. It doesn't feel natural. I think its the "Would you" part that I'm tripping up on. Maybe something like "Would you mind answering a question..."

The problem with the question "Would you mind" is that "sure" is technically a response that means "leave me alone, I mind you bugging me".

What about "Have a second to answer one question about WordPress.com?"?

**Comment by mattsherman** [comment]
Thanks for the review @jonathansadowski -- I agree that we can do further wording tweaks in a follow-up PR, since this feature isn't live yet.

**Comment by mjuhasz** [comment]
Good job 👍

Jon tested it in Chrome (desktop and mobile), Firefox, and IE11, and I can confirm that it works well in Microsoft Edge and Safari as well.

**Pull request merged**